src/firefly/js/ui/FileUploadProcessor.jsx

```diff
@@ -0,0 +1,197 @@
+import {FileAnalysisType, Format} from 'firefly/data/FileAnalysis';
+import {showInfoPopup} from 'firefly/ui/PopupUtil';
+import {getAppHiPSForMoc, isMOCFitsFromUploadAnalsysis} from 'firefly/visualize/HiPSMocUtil';
+import {MetaConst} from 'firefly/data/MetaConst';
+import {isLsstFootprintTable} from 'firefly/visualize/task/LSSTFootprintTask';
+import {makeFileRequest, makeTblRequest} from 'firefly/tables/TableRequestUtil';
+import {getAppOptions} from 'firefly/core/AppDataCntlr';
+import {getFieldVal} from 'firefly/fieldGroup/FieldGroupUtils';
+import {getFileFormat, getFirstPartType, getSelectedRows, isRegion, panelKey} from 'firefly/visualize/ui/FileUploadViewPanel';
```

Comment:
You should not be importing anything from `FileUploadViewPanel.jsx`. Think of `FileProcessor` being the utility routines. You should move these functions of to `FileProcessor` and import them the other way. 

`panelKey` should no longer exist. That should be the group key that is passed, it not be hardcoded.